pkg/v5/configmap/desired_test.go

```diff
@@ -0,0 +1,111 @@
+package configmap
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/giantswarm/micrologger/microloggertest"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func Test_ConfigMap_GetDesiredState(t *testing.T) {
+	testCases := []struct {
+		name               string
+		configMapValues    ConfigMapValues
+		expectedConfigMaps []*corev1.ConfigMap
+	}{
+		{
+			name: "case 0: basic match",
+			configMapValues: ConfigMapValues{
+				ClusterID:    "5xchu",
+				Organization: "giantswarm",
+				WorkerCount:  3,
+			},
+			expectedConfigMaps: []*corev1.ConfigMap{
+				&corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "nginx-ingress-controller-values",
+						Namespace: metav1.NamespaceSystem,
+						Labels: map[string]string{
+							"app": "nginx-ingress-controller",
+							"giantswarm.io/cluster":      "5xchu",
+							"giantswarm.io/managed-by":   "cluster-operator",
+							"giantswarm.io/organization": "giantswarm",
+							"giantswarm.io/service-type": "managed",
```

Comment:
I am wondering why we don't use the label constants here.

Reply:
Good call. Done.